CMakeLists.txt

```diff
@@ -0,0 +1,613 @@
+CMAKE_MINIMUM_REQUIRED (VERSION 2.8)
+INCLUDE (CheckIncludeFiles)
+INCLUDE (CheckTypeSize)
+INCLUDE (CheckFunctionExists)
+INCLUDE (CheckSymbolExists)
+INCLUDE (CheckCSourceCompiles)
+INCLUDE (CheckStructHasMember)
+INCLUDE (CheckLibraryExists)
+
+PROJECT (cares C)
+
+# This is for libtool compatibility, and specified in a form that is easily
+# translatable from libtool (even if the actual form doesn't make sense).
+# For instance, in an autotools project, in Makefile.am there is a line that
+# contains something like:
+#     -version-info 4:0:2
+# This breaks down into sections of current:revision:age
+# This then generates a version of  "(current-age).age.revision"  with an
+# interface version of "(current-age)"
+# For example, a version of 4:0:2 would generate output such as:
+#    libname.so   -> libname.so.2
+#    libname.so.2 -> libname.so.2.2.0
+SET (CARES_LIB_VERSIONINFO "4:0:2")
+
+
+OPTION (CARES_STATIC     "Build as a static library"                                             OFF)
+OPTION (CARES_SHARED     "Build as a shared library"                                             ON)
+OPTION (CARES_INSTALL    "Create installation targets (chain builders may want to disable this)" ON)
+OPTION (CARES_STATIC_PIC "Build the static library as PIC (position independent)"                OFF)
+
+
+# Keep build organized.
+SET (CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
+SET (CMAKE_LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
+SET (CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
+SET (PACKAGE_DIRECTORY ${PROJECT_BINARY_DIR}/package)
+
+
+# Install path helpers
+IF ("${CARES_INSTALL_LOCATION_LIBS}" STREQUAL "")
+	SET (CARES_INSTALL_LOCATION_LIBS ${CMAKE_INSTALL_PREFIX}/lib CACHE INTERNAL "")
+ENDIF ()
+IF ("${CARES_INSTALL_LOCATION_HEADERS}" STREQUAL "")
+	SET (CARES_INSTALL_LOCATION_HEADERS ${CMAKE_INSTALL_PREFIX}/include CACHE INTERNAL "")
+ENDIF ()
+IF ("${CARES_INSTALL_LOCATION_BIN}" STREQUAL "")
+	SET (CARES_INSTALL_LOCATION_BIN ${CMAKE_INSTALL_PREFIX}/bin CACHE INTERNAL "")
+ENDIF ()
+
+
+# When chain building C-Ares, we should set some of the same variables that a 
+# standard Find_Package(CAres) would set.  It just references the internal
+# build paths rather than an already-installed version.
+SET (CARES_FOUND 1 CACHE INTERNAL "CARES LIBRARY FOUND")
```

Comment:
Exactly :)  I actually chain build it with another project (infact, thats also why I have a static pic option so it can be embedded into another .so) ... so I have it output the same variables as a FindPackage type module would so I can see if the directory exists within the project, if it does chain build it, otherwise call FindPackage ... and everything sets the same vars :)